src/main/duckdb-c.cpp

```diff
@@ -1045,7 +1045,11 @@ duckdb_state duckdb_appender_run_function(duckdb_appender appender, FUN &&functi
 	auto *appender_instance = (Appender *)appender;
 	try {
 		function(*appender_instance);
+	} catch (std::exception &ex) {
+		appender_instance->error = ex.what();
```

Comment:
This appears to be uncovered, could you add an appender test that triggers this error?